src/dodal/devices/zocalo/zocalo_results.py

```diff
@@ -56,12 +56,25 @@ def bbox_size(result: XrcResult):
     ]
 
 
+# TODO this may give uneccessary warnings if dicts are different within rounding errors
+def get_dict_differences(
```

Comment:
This will be sensitive to rounding errors, so I could change the function to something a bit more complicated to avoid that

Reply:
I couldn't find any function which did something like math.isclose which worked for this kind of dictionary, so this was me being lazy instead. I think iterating through each key and treating lists differently to floats is the way to do this properly

Reply:
It looks like `deepdiff` https://zepworks.com/deepdiff/current/diff.html might do this with `math_epsilon` being the tolerance. I think on balance it's fine to add another dependency but given this check is probably temporary add a comment next to the dependency pointing to removing it in a ticket that around cleaning up GPU stuff

Reply:
Ah, nice!